File: morris/morris_game/game_gui.py
#!/usr/bin/env python
'''
Created on Oct 3, 2015

@author: speng
'''
# from morris_game.Game_controller import Game_controller, Game_state
import re
from Game_controller import Game_controller
from Game_controller import Game_state
from Position import Position
# from morris_game.Position import Position

def parse_input(user_input):
    pat = r"\(\s*\d\s*,\s*\d\s*\)"
    match = re.search(pat, user_input)
    if match is not None:
        return eval(match.group())
    return None

def get_input_coord():
    #coord = input("enter a postion in format (x,y)\n")
    user_input = raw_input("enter a postion in format (x,y)\n")
    coord = parse_input(user_input)
    while coord is None:
        print "invalid input"
        user_input = raw_input("enter a postion in format (x,y)\n")
        coord = parse_input(user_input)
    return coord

if __name__ == '__main__':
    game = Game_controller()
    AI = False
    AI_player = game.player_2
    while(game.state != Game_state.player_1_win and game.state != Game_state.player_2_win):
        print game.board
        print game.current_player
        print game.state
        if AI == True and game.current_player == AI_player:
            #randomly generate a coordinate
            pass
        else:
            coord = get_input_coord()
            pos = Position(coord,game.current_player)
        if(not game.move(pos)):
            print "illegal move"
    print game.board
    print game.state
    
        

File: morris_gui/src/morris_java/GameGui.java
package morris_java;

import javax.swing.JFrame;

public class GameGui {
	public void start() throws InterruptedException{
		JFrame f = new JFrame("Nine Men's Morris");
		f.setSize(800, 900);
		f.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		
		GameController game = new GameController();
		boolean AI = false;
		Player AI_player = game.player_1;
		
		f.add(game.guiboard);
		
		f.setResizable(false);
		f.setVisible(true);
		
		while (game.state != GameState.player_1_win
				&& game.state != GameState.player_2_win) {
			if(game.guiboard.getClickPoint() > 24){
				Thread.currentThread();
				Thread.sleep(1);
				continue;
			}
			System.out.println(game.board);
			System.out.println(game.current_player);
			System.out.println(game.state);
			if (AI && game.current_player == AI_player) {
				// TODO randomly generate a coordinate
			} else {
				Coord coord = getInputCoord(game.guiboard);
				
				Position pos = new Position(coord, game.current_player);
				if (!game.move(pos)) {
					System.out.println("illegal move");
				}
				else {
					if(game.state == "move_2" || game.state == "fly_2")
						game.selectTokenOnGui(coord);
					else
						game.updateGui();
				}
			}
		}
		System.out.println(game.board);
		System.out.println(game.state);
	}
	
	private Coord[] coordinate = {null, new Coord(0,0), new Coord(0,3), new Coord(0,6),
			                            new Coord(3,6), new Coord(6,6), new Coord(6,3),
			                            new Coord(6,0), new Coord(3,0), new Coord(1,1),
			                            new Coord(1,3), new Coord(1,5), new Coord(3,5),
			                            new Coord(5,5), new Coord(5,3), new Coord(5,1),
			                            new Coord(3,1), new Coord(2,2), new Coord(2,3),
			                            new Coord(2,4), new Coord(3,4), new Coord(4,4),
			                            new Coord(4,3), new Coord(4,2), new Coord(3,2)};

	private Coord getInputCoord(GuiBoard b) {
		Coord userInput = null;
		
		userInput = coordinate[b.getClickPoint()];
		// in.close();
		
		b.setCLickPoint();
		return userInput;
	}

}
